CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+* If you're not sure if your issue is the same, err on the side of caution and file your issue.
+  You can add a comment to include the issue number (and link) for the similar issue. If we evaluate
+  your issue as being the same as the existing issue, we'll close the duplicate.
+* If your issue doesn't exist, use the issue template to file a new issue.
+  * When filing an issue, be sure to provide as much information as possible, including script output so
+    we can collect information about your configuration. This helps reduce the time required to
+    reproduce your issue.
+  * Check your issue regularly, as we may require additional information to successfully reproduce the
+    issue.
+* You may also open an issue to ask questions to the maintainers about whether a proposed change
+  meets the acceptance criteria, or to discuss an idea pertaining to the library.
+
+## Acceptance Criteria ##
+
+The purpose of hipCUB is to provide a thin wrapper library on top of rocPRIM or CUB. This wrapper allows users to port CUB projects to HIP so that they can also be run on AMD hardware.
+
+Because it is a wrapper, the implementations of the algorithms that hipCUB API calls refer to are not contained within hipCUB.
+Rather, they exist within rocPRIM and CUB - the hipCUB API functions simply delegate the work to these underlying libraries.
+This delegation should be performed in a manner that minimizes overhead.
+
+When a pull request is created, a number of automated checks are run. These checks:
+- test the change on various OS platforms (Ubuntu, RHEL, etc.)
+- run on different GPU architectures (MI-series, Radeon series cards, etc.)
+- run benchmarks to check for performance degredation
+
+In order for change to be accepted:
+- it must pass all of the automated checks
+- it must undergo a code review
+
+The GitHub "Issues" tab may also used to discuss ideas surrounding particular features or changes before raising pull requests.
+
+## Code Structure ##
+
+hipCUB is a header-only library. Library code lives inside of /hipcub/include.
+Code is organized by the level-of-scope at which it operates. For example, the following
+subdirectories are organized by hardware-level scope:
+* device/ contains headers for device-level algorithms
+* block/ contains headers for block-level algorithms
+* warp/ contains headers for warp/wavefront-level algorithms
+
+The following subdirectories are organized according to software-scope level:
+* grid/ contains headers for grid-level operations (barriers, queues, etc.)
+* thread/ contains headers for thread-level operations (load/store, scan, reduce, etc.)
+
+Finally, the iterator/ subdirectory provides access to the iterators that are used to interact with most algorithms in the library.
+
+Back at the root level, you can find tests and benchmarks located inside directories of the same name. 
+There is also an examples folder that contains a number of sample API use cases.
+
+## Coding Style ##
```

Comment:
This section is copied from rocSPARSE's CONTRIBUTING.md. I believe most of this holds true for us here in hipCUB as well, but I'm not 100% certain. If you see anything you think should be modified, feel free to let me know.

Reply:
I made minor edits to your text.